### DISCUSSION TITLE
May I post my solutions for codewars tasks on github?

### DISCUSSION THREAD

**Discussion by Oleg83-IT** [codewars/codewars.com#2590] [General]
Hi there

I'd like to solve tasks on codewars and post my solutions on my github page to demonstrate my skills to someone, who can hire me. I don't know, is it permitted or not by codewars policy. It will look like this: 

Task from codewars: ...
My solution: ...

Is this permitted by codewars policy?

**Comment by hobovsky** [comment]
Please see the last point of Rules in Codewars Code of Conduct: https://docs.codewars.com/community/rules#rules .

**Reply by Oleg83-IT** [reply]
> Please see the last point of Rules in Codewars Code of Conduct: https://docs.codewars.com/community/rules#rules .

Got it, I will not do so, but just wondering what is the penalty for such behavior?

**Reply by DonaldKellett** [reply]
Posting solutions publicly is highly discouraged but generally not penalized on the Codewars end (though there might be exceptions). Plagiarizing solutions, on the other hand, will at least result in your rank, honor and eligibility for affected Kata revoked, and in serious cases result in a permanent ban where your account's rank and honor is pinned to `8 kyu / 0%` indefinitely with no way of deleting your account.

**Reply by hobovsky** [reply]
> what is the penalty for such behavior?

Currently it's only a Codewars mod raising an issue in your repository kindly asking to make it private. There's no real penalty applied, except some potential backlash from community dissatisfied with publication of spoilers.

I know there are some sites which reserve themselves rights to penalize users much more severely. For example Project Euler changed their policy some time last year or two, and now they can eventually lock an account of a person who publishes solutions of difficult problems. Codewars does no do this (for now ;) ) but publication of spoilers, especially for difficult kata, is strongly discouraged and generally frowned upon.

**Reply by Oleg83-IT** [reply]
Thanks for the explanation. Does anyone know any sites, where it is not frowned upon? Where I can take a task, solve it and post it as my portfolio?

**Reply by jhoffner** [reply]
To be honest, employers don't care. If they are looking at Github, they want to see real projects. There is no reason for them to even believe you wrote that code. What stops someone from just posting random code into a Github repo? That's if they look at all, top of funnel recruiters don't know enough about coding to make any sense of Github.

I would link your Codewars profile somewhere that employers can see, so that they see that you have put in a lot of time and energy. They will understand the passion and effort, even if they aren't technical. Of course, you should prioritize any real-world work you have done first, and include Codewars more as an additional signal.

**Reply by Oleg83-IT** [reply]
> Of course, you should prioritize any real-world work you have done first, and include Codewars more as an additional signal.

Sounds reasonable, thank you Jake!
